Kubernetes + Smallrye Health Extensions: Knative deployment fails due to presence of startupProbe

### Describe the bug

The Smallrye health extension adds a startupProbe to the generated target/kubernetes/knative.yml when I set my deployment-target to knative.  The Knative admission controller however does not allow the startupProbe element, resulting in a failed deployment.   

### Expected behavior

I expect the knative.yml to not contain the unsupported startupProbe element.

### Actual behavior

_No response_

### How to Reproduce?

Steps to reproduce:

1. quarkus create app knative-bug -x kubernetes -x smallrye-health
2. mvn package -Dquarkus.kubernetes.deployment-target=knative
3. verify that a startupProbe is present in the generated target/knative.yml
4. deploy to Knative, and observe the following error:

```
[ERROR] Caused by: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://xyz.openshiftapps.com:6443/apis/serving.knative.dev/v1/namespaces/blabla/services. Message: admission webhook "validation.webhook.serving.knative.dev" denied the request: validation failed: must not set the field(s): spec.template.spec.containers[0].startupProbe. Received status: Status(apiVersion=v1, code=400, details=null, kind=Status, message=admission webhook "validation.webhook.serving.knative.dev" denied the request: validation failed: must not set the field(s): spec.template.spec.containers[0].startupProbe, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=BadRequest, status=Failure, additionalProperties={}).
```


### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java
+++ b/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesCommonHelper.java
@@ -4,6 +4,7 @@ package io.quarkus.kubernetes.deployment;
 import static io.dekorate.kubernetes.decorator.AddServiceResourceDecorator.distinct;
 import static io.quarkus.kubernetes.deployment.Constants.DEFAULT_HTTP_PORT;
 import static io.quarkus.kubernetes.deployment.Constants.HTTP_PORT;
+import static io.quarkus.kubernetes.deployment.Constants.KNATIVE;
 import static io.quarkus.kubernetes.deployment.Constants.QUARKUS_ANNOTATIONS_BUILD_TIMESTAMP;
 import static io.quarkus.kubernetes.deployment.Constants.QUARKUS_ANNOTATIONS_COMMIT_ID;
 import static io.quarkus.kubernetes.deployment.Constants.QUARKUS_ANNOTATIONS_VCS_URL;
@@ -1012,7 +1013,9 @@ public class KubernetesCommonHelper {
         List<DecoratorBuildItem> result = new ArrayList<>();
         createLivenessProbe(name, target, livenessProbe, livenessPath).ifPresent(d -> result.add(d));
         createReadinessProbe(name, target, readinessProbe, readinessPath).ifPresent(d -> result.add(d));
-        createStartupProbe(name, target, startupProbe, startupPath).ifPresent(d -> result.add(d));
+        if (!KNATIVE.equals(target)) { // see https://github.com/quarkusio/quarkus/issues/33944
+            createStartupProbe(name, target, startupProbe, startupPath).ifPresent(d -> result.add(d));
+        }
         return result;
     }
 

--- a/integration-tests/kubernetes/maven-invoker-way/src/it/knative-jib-build-and-deploy/pom.xml
+++ b/integration-tests/kubernetes/maven-invoker-way/src/it/knative-jib-build-and-deploy/pom.xml
@@ -38,6 +38,10 @@
     </dependency>
     <dependency>
       <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-smallrye-health</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.quarkus</groupId>
       <artifactId>quarkus-junit5</artifactId>
       <scope>test</scope>
     </dependency>

--- a/integration-tests/kubernetes/quarkus-standard-way/pom.xml
+++ b/integration-tests/kubernetes/quarkus-standard-way/pom.xml
@@ -38,6 +38,10 @@
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-smallrye-health</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
             <artifactId>quarkus-junit5-internal</artifactId>
             <scope>test</scope>
         </dependency>
@@ -223,6 +227,19 @@
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-container-image-s2i-deployment</artifactId>
+            <version>${project.version}</version>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-smallrye-health-deployment</artifactId>
             <version>${project.version}</version>
             <type>pom</type>
             <scope>test</scope>

--- a/integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/KnativeWithHealthTest.java
+++ b/integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/KnativeWithHealthTest.java
@@ -71,13 +71,7 @@ public class KnativeWithHealthTest {
                                     assertNotNull(p.getHttpGet());
                                     assertNull(p.getHttpGet().getPort());
                                 });
-                                assertThat(c.getStartupProbe()).isNotNull().satisfies(p -> {
-                                    assertThat(p.getInitialDelaySeconds()).isEqualTo(5);
-                                    assertProbePath(p, "/q/health/started");
-
-                                    assertNotNull(p.getHttpGet());
-                                    assertNull(p.getHttpGet().getPort());
-                                });
+                                assertThat(c.getStartupProbe()).isNull();
                             });
                         });
                     });

--- a/integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/KnativeWithHealthUsingManagementInterfaceTest.java
+++ b/integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/KnativeWithHealthUsingManagementInterfaceTest.java
@@ -71,6 +71,7 @@ public class KnativeWithHealthUsingManagementInterfaceTest {
                                     assertNotNull(p.getHttpGet());
                                     assertNull(p.getHttpGet().getPort());
                                 });
+                                assertThat(c.getStartupProbe()).isNull();
                             });
                         });
                     });

